mlflow/tracking/_tracking_service/client.py

```diff
@@ -255,11 +258,23 @@ def _record_logged_model(self, run_id, mlflow_model):
         self.store.record_logged_model(run_id, mlflow_model)
 
     def _get_artifact_repo(self, run_id):
```

Comment:
@apurva-koti Can we add a unit test here to ensure that the same object is fetched every time `_get_artifact_repo` is called with a particular run ID?

Reply:
Done!